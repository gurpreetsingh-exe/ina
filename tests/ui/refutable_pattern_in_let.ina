// COMPILE_FAIL

type T = A | B
type T2 = A

fn main() {
    // ok
    let T2::A = T2::A;
    // error
    let T::A = T::B;
}
