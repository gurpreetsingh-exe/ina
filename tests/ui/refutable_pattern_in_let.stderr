error: refutable pattern in let binding
    ╭─[tests/ui/refutable_pattern_in_let.ina:10:5]
    │
 10 │     let T::A = T::B;
    │     ────────┬───────
    │             ╰───────── pattern `B` not covered
    ∙
────╯
